### PR TITLE
add sdk version on http header.

### PR DESCRIPTION
**Pull request by rizakii**
Add SDK information on HTTP Request header.

### CODE REVIEW

**Comment by satoshikumano** [comment]
What if it is used by thing-if-ThingSDK?
https://github.com/KiiPlatform/thing-if-ThingSDK/pull/38

Would have two x-kii-sdk header?
I would like to avoid having two x-kii-sdk header.

File: kii-core/kii_core.c [hunk 1]
```diff
@@ -110,6 +111,9 @@ kii_core_init(
 
     kii->app_id = (char*)app_id;
     kii->app_key = (char*)app_key;
+    kii_sprintf(kii->sdk_info,
+            "sn=is;sv=%s;",
```
Comment (satoshikumano) [inline kii-core/kii_core.c]: sn=te

**Comment by rizakii** [comment]
@satoshikumano 
>>Would have two x-kii-sdk header?
>>I would like to avoid having two x-kii-sdk header.

It won't duplicated since thing if rewrite the header on initialisation.

[This line](https://github.com/KiiPlatform/thing-if-ThingSDK/pull/38/files#diff-c258ab5fc1373e5b397a1eec42f71cf2R134) rewrite the header. I would like to add the embedded sdk version on it. 

Thats why the version string is defined on public header.

**Comment by shinsuke-nara** [comment]
@satoshikumano @rizakii In my environment, sample application does not work well. It halts with segmentation fault. I think this PR should not be merge.

I will fix this. Please wait.

**Comment by shinsuke-nara** [comment]
@satoshikumano @rizakii The bug is in following line.

https://github.com/KiiPlatform/KiiThingSDK-Embedded/pull/107/commits/2701d51728ad5cf45e1420ff996418711828e242#diff-76cdc866c355e50bad70eb5345189c3eR114

The code is like following:

```c
kii_sprintf(kii->sdk_info,
        "sn=is;sv=%s;",
        KII_SDK_VERSION);
```

sdk_info is char pointer. That is not array of char. So this code assign string to untouchable memory.

I fixed that bug. We can merge this PR.

Could you  review this?

File: tests/large_test/Makefile [hunk 2]
```diff
@@ -40,7 +40,7 @@ $(KIICLIENT):
 	cp $(KIICLIENT) ./
 
 $(KIISDK):
-	$(MAKE) -C ../../kii $(KIISDKPARAM) "CFLAGS+=-DKII_JSON_FIXED_TOKEN_NUM=256"
+	$(MAKE) -C ../../kii $(KIISDKPARAM) "CFLAGS+=-DKII_JSON_FIXED_TOKEN_NUM=256 -DKII_PUSH_KEEP_ALIVE_INTERVAL_SECONDS=30"
```
Comment (rizakii) [inline tests/large_test/Makefile]: I think this is out of scope. And I think, compiler flag better mentioned on README.

**Comment by shinsuke-nara** [comment]
@satoshikumano I have removed `sdk_info` fields. Instead of `sdk_info`, I have added two header files.
These header files are same name which is `kii_sdk_version.h`. These header files contain sdk version information by `define` macro.

Could you review this?

File: kii-core/kii_core.h [hunk 3]
```diff
@@ -424,6 +424,9 @@ typedef struct kii_core_t
     kii_state_t _state;
 
     size_t _content_length;
+
+    /** Kii Cloud sdk info */
+    const char* sdk_info;
```
Comment (satoshikumano) [inline kii-core/kii_core.h]: _ sdk_info

File: kii-core/kii_core.c [hunk 4]
```diff
@@ -66,15 +67,17 @@
 #define HTTP1_1 "HTTP/1.1 "
 #define END_OF_HEADER "\r\n\r\n"
 #define CONST_LEN(str) sizeof(str) - 1
+#define KII_SDK_VERSION "sn=tec;sv=1.1.1"
```
Comment (satoshikumano) [inline kii-core/kii_core.c]: KII_SDK_INFO

File: kii-core/kii_core.c [hunk 5]
```diff
@@ -66,15 +67,17 @@
...
 
 const char DEFAULT_OBJECT_CONTENT_TYPE[] = "application/json";
 
     kii_error_code_t
-kii_core_init(
+kii_core_init_with_version(
```
Comment (satoshikumano) [inline kii-core/kii_core.c]: kii_core_init_with_info

File: kii-core/kii_core.c [hunk 6]
```diff
@@ -66,15 +67,17 @@
...
         kii_core_t* kii,
         const char* site,
         const char* app_id,
-        const char* app_key)
+        const char* app_key,
+        const char* version)
```
Comment (satoshikumano) [inline kii-core/kii_core.c]: version -> sdk_info

**Comment by shinsuke-nara** [comment]
This PR is moved to #108
